# Characterization of emerging swine viral diseases through Oxford Nanopore sequencing 

This repository provides relavant bioinformatics methods and codes for the manuscript: Characterization of emerging swine viral diseases through Oxford Nanopore sequencing using SVA as a model

The objective of this study is to provide hands-on reference for viral infectious disease investigation, especially under emering situations.

## Sequencing and raw reads assessment

Basecalling of raw reads was performed using Guppy (https://nanoporetech.com/) to generate FASTQ files. 

Total yield, total reads, read quality, and read length from whole genome sequencing were analyzed using NanoPlot (https://github.com/wdecoster/NanoPlot).

In order to evaluate the raw error rates of direct RNA sequencing, raw reads were mapped to the VR2332 reference sequence using minimap2-2.12 (https://github.com/lh3/minimap2), processed with SAMtools (https://github.com/samtools/samtools) to generate BAM files, and then evaluated by AlignQC (https://github.com/jason-weirather/AlignQC).

```
minimap2 -ax map-ont prrsv.fastq > minimap_prrsv.sam  

samtools view -b minimap_prrsv.sam > minimap_prrsv.bam
samtools sort minimap_prrsv.bam -o minimap_prrsv_sorted.bam
samtools index minimap_prrsv_sorted.bam

qualimap_v2.2.1/qualimap bamqc -bam minimap_prrsv_sorted.bam -outdir qualimap_results

alignqc analyze minimap_prrsv_sorted.bam -r reference.fasta --no_annotation -o prrsv_alignqc.xhtml
```
## Species detection

## Strain detection

a custom SVA sequence database containing 951 PRRSV whole genome sequences was generated by downloading all PRRSV whole genome sequences available in GenBank (949 sequences including our VR2332 strain, download date: Nov 2018) with the addition of sequences from our SDEU (MN175678) and 1-7-4 (MN175677) lab strains. 
Alignment Search Tool (BLAST) with a significance filter of expect value (E) < 10 ^-50 to examine the PRRSV sequence reads. 

```
makeblastdb -in downloaded_prrsv.fasta -title prrsvdatabase -dbtype nucl -out prrsvdatabase
blastn -db prrsvdatabase -query prrsv.fastq -evalue 1e-50 -word_size 11 -outfmt 7 > prrsv_blast
```

## Whole genome generation

DRS: Software and version: Racon-1.3.2

```
cat prrsv.fastq | paste - - - - | awk -F '\t' '{L=length($2);if(L>M) {M=L;R=$0;}} END {print R;}' | tr "\t" "\n" > largest.fastq

seqtk-master/seqtk seq -a largest.fastq > largest.fasta

minimap2 -ax map-ont largest.fasta prrsv.fastq > prrsv_mapped.sam  

racon prrsv.fastq prrsv_mapped.sam largest.fasta > racon.fasta
```

PCS: Software and version: Canu-1.6 (https://github.com/marbl/canu)

```
canu -p ssuisX -d ssuis__assembly genomeSize=2m -nanopore-raw ssuisX.fastq useGrid=0 # using MinION raw reads ssuisX.fastq to generate assembly ssuisX.contigs.fasta.
```
